include/word.h

```diff
@@ -15,14 +18,18 @@
 bool valid_word(dict_t *dict, char* word);
 
 /*
- * generate_suggestions: receives a dictionary, word, and an allocated array of suggestions
- * fills array of suggestions with words
- * note that this is currently hard-coded;
- * parameters: word, dictionary, array list to which suggestions are stored
- * returns: int (EXIT_SUCCESS OR FAILURE). FAILURE if no suggestions generated
+ * generate_suggestions - Returns a list of possible suggestions
+ *
+ * Parameters:
+ *  - dict: A dictionary pointer. Must point to already allocated memory.
+ *  - word: A string (word).
+ *  - suggestions: An array of strings (words).
+ *  - max_edits: Number of maximum edits.
+ *  - amount: Amount of suggestions.
+ *
+ * Returns:
+ *  - int: EXIT_SUCCESS or EXIT_FAILURE (EXIT_FAILURE if no suggestions generated.)
```

Comment:
Let's make this function return the list of suggestions, rather than using an out-parameter and returning a success code, and have `NULL` be the failure case.

Reply:
Ok, changed!